### DISCUSSION TITLE
How do I setup up my workflow?

### DISCUSSION THREAD

**Discussion by zetashift** [chipsenkbeil/org-roam.nvim#40] [Q&A] [answered]
So, I'm new to orgmode, as well as `nvim-orgmode`. I used linking heavily in my `LogSeq` setup, and was wondering how I should get started.

Currently I have the following `nvim-orgmode` setup:
```lua
local org_path = function(path)
  local org_directory = '~/self'
  return ('%s/%s'):format(org_directory, path)
end

local orgmode_config = {
  org_agenda_files = org_path '**/*',
  org_default_notes_file = org_path 'refile.org',
  org_hide_emphasis_markers = true,
  org_agenda_text_search_extra_files = { 'agenda-archives' },
  org_agenda_start_on_weekday = false,
  org_startup_indented = true,
  org_log_into_drawer = 'LOGBOOK',
  org_todo_keywords = { 'TODO(t)', 'PROGRESS(p)', '|', 'DONE(d)', 'REJECTED(r)' },
  org_capture_templates = {
    t = {
      description = 'Refile',
      template = '* TODO %?\nDEADLINE: %T',
    },
    w = {
      description = 'Todo',
      template = '* TODO %?\nDEADLINE: %T',
      target = org_path 'todos.org',
    },
    j = {
      description = 'Journal',
      template = '* %?',
      target = org_path 'journal.org',
      datetree = { reversed = true },
    },
  },
  mappings = {
    prefix = '<leader>n',
  },
}

return {
  'nvim-orgmode/orgmode',
  event = 'VeryLazy',
  dependencies = {
    { 'nvim-orgmode/org-bullets.nvim' },
  },
  ft = { 'org', 'orgagenda' },
  config = function()
    require('orgmode').setup(orgmode_config)
  end,
}
```

However this is quite different from `org-roam.nvim`s setup. My questions are:
- Can I set `org-roam.nvim` as a dependency and things will just work?
- I installed `org-roam.nvim` without my config, and saw that there was heavy usage of daily note in separate files. I currently prefer using a single `journal.org` file and a `datetree`. Does roam play well with this? Or should I switch over to separate files.

Thank you for this cool project!

**Comment by chipsenkbeil** [comment, marked as answer]
Hey there! The dailies extension makes use of separate files named by date. So if you want date functionality with org roam then you'd need to switch to its structure.

However, the standard org roam nodes can be placed at file or heading level. So, you could leverage capture to create your journal entries, and add a properties drawer underneath with an ID. Then roam would be able to treat them as nodes.

```org
* My Entry on May 18, 2024
   :PROPERTIES:
   :ID: abc
   :END:
* My Entry on May 19, 2024
   :PROPERTIES:
   :ID: def
   :END:
```

Keep in mind calendar features or anything from dailies wouldn't work here.

**Reply by zetashift** [reply]
Thank you! Since I like the calendar feature, I'll keep the structure of `dailies`. I liked the easy browsing of a list of days in the `journal.org` setup, but it's not that pressing.
 
The `lazy.nvim` config in the end looks like this:
```
return {
  {
    'nvim-orgmode/orgmode',
    event = 'VeryLazy',
    dependencies = {
      { 'nvim-orgmode/org-bullets.nvim' },
    },
    ft = { 'org', 'orgagenda' },
    config = function()
      require('orgmode').setup(orgmode_config)
    end,
  },
  {
    'chipsenkbeil/org-roam.nvim',
    tag = '0.1.0',
    config = function()
      require('org-roam').setup {
        directory = org_path '',
      }
    end,
  },
}
```
EDIT: I should probably pin `orgmode` as well.

**Reply by chipsenkbeil** [reply]
Yeah, tag orgmode to 0.3.4 and add it as a dependency to org roam to make sure they load it the correct order.

**Reply by zetashift** [reply]
That would bring it to:
```lua
return {
  {
    'nvim-orgmode/orgmode',
    event = 'VeryLazy',
    tag = '0.3.4',
    dependencies = {
      { 'nvim-orgmode/org-bullets.nvim' },
    },
    ft = { 'org', 'orgagenda' },
    config = function()
      require('orgmode').setup(orgmode_config)
    end,
  },
  {
    'chipsenkbeil/org-roam.nvim',
    tag = '0.1.0',
    dependencies = {
      {
        'nvim-orgmode/orgmode',
      },
    },
    config = function()
      require('org-roam').setup {
        directory = org_path '',
      }
    end,
  },
}
```

**Reply by chipsenkbeil** [reply]
I don't know how lazy works, regarding whether it matters or not to specify the tag in the dependencies section as well. Otherwise, seems like it works fine! This should create a `daily` directory within `~/self` since you set the roam directory to be the same as your org directory.
